test/test.py

```diff
@@ -1,20 +1,107 @@
-# SPDX-FileCopyrightText: © 2024 Tiny Tapeout
-# SPDX-License-Identifier: Apache-2.0
-
 import cocotb
 from cocotb.clock import Clock
 from cocotb.triggers import ClockCycles
+from cocotb.triggers import RisingEdge, Timer
 
 
-@cocotb.test()
-async def test_project(dut):
-    dut._log.info("Start")
+# ---------------- SPI bit-bang helpers ----------------
+
+class SpiCfg:
+    def __init__(self, cpol=False, cpha=False, msb_first=True, sclk_period_ns=40):
+        self.cpol = int(bool(cpol))
+        self.cpha = int(bool(cpha))
+        self.msb_first = bool(msb_first)
+        self.t_half = sclk_period_ns // 2  # ns
+
+
+async def spi_transfer_byte(dut, cfg: SpiCfg, sclk, mosi, miso, cs_n, tx_byte: int) -> int:
+    """
+    Transfer one 8-bit word with one CS pulse.
+    Returns the received byte.
+    CPOL/CPHA modes:
+      - Mode 0: cpol=0,cpha=0  (sample on rising)
+      - Mode 1: cpol=0,cpha=1  (sample on falling)
+      - Mode 2: cpol=1,cpha=0  (sample on falling)
+      - Mode 3: cpol=1,cpha=1  (sample on rising)
+    """
+    # Idle levels
+    sclk.value = cfg.cpol
+    cs_n.value = 1
+    mosi.value = 0
+    await Timer(cfg.t_half, units="ns")
+
+    # Assert CS for this byte
+    cs_n.value = 0
+    await Timer(cfg.t_half, units="ns")
+
+    rx = 0
+    bit_indices = range(7, -1, -1) if cfg.msb_first else range(0, 8)
```

Comment:
So sampling on rising edge mode 0 let's consider this 

1. Assert expected answer for given input
2. Make sure some unwanted state is not reached either in the cordic core or from the spi slave. 
3. We need to define invariants for this first.